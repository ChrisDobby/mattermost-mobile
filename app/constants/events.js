// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    SET_CHANNEL_LOADING: 'set_channel_loading',
    SWITCH_CHANNEL: 'switch_channel',
};
