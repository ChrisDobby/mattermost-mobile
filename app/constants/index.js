// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import Posts from './posts';
import PermissionTypes from './permissions';
import ViewTypes, {UpgradeTypes} from './view';

import General from './general';
import Preferences from './preferences';

export {
    General,
    Preferences,
    Posts,
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    PermissionTypes,
    UpgradeTypes,
    ViewTypes,
};
