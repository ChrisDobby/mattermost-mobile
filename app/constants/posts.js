// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const PostTypes = {
    CHANNEL_DELETED: 'system_channel_deleted',
    DISPLAYNAME_CHANGE: 'system_displayname_change',
    CONVERT_CHANNEL: 'system_convert_channel',
    EPHEMERAL: 'system_ephemeral',
    EPHEMERAL_ADD_TO_CHANNEL: 'system_ephemeral_add_to_channel',
    HEADER_CHANGE: 'system_header_change',
    PURPOSE_CHANGE: 'system_purpose_change',

    JOIN_LEAVE: 'system_join_leave',
    JOIN_CHANNEL: 'system_join_channel',
    LEAVE_CHANNEL: 'system_leave_channel',
    ADD_REMOVE: 'system_add_remove',
    ADD_TO_CHANNEL: 'system_add_to_channel',
    REMOVE_FROM_CHANNEL: 'system_remove_from_channel',

    JOIN_TEAM: 'system_join_team',
    LEAVE_TEAM: 'system_leave_team',
    ADD_TO_TEAM: 'system_add_to_team',
    REMOVE_FROM_TEAM: 'system_remove_from_team',

    COMBINED_USER_ACTIVITY: 'system_combined_user_activity',
    ME: 'me',
    ADD_BOT_TEAMS_CHANNELS: 'add_bot_teams_channels',
};

const PostListTypes = {
    COMBINED_USER_ACTIVITY: 'user-activity-',
    DATE_LINE: 'date-',
    START_OF_NEW_MESSAGES: 'start-of-new-messages',
};

export default {
    MAX_COMBINED_SYSTEM_POSTS: 100,
    POST_LIST_TYPES: PostListTypes,
    POST_TYPES: PostTypes,
    IGNORE_POST_TYPES: [
        PostTypes.ADD_REMOVE,
        PostTypes.ADD_TO_CHANNEL,
        PostTypes.CHANNEL_DELETED,
        PostTypes.JOIN_LEAVE,
        PostTypes.JOIN_CHANNEL,
        PostTypes.LEAVE_CHANNEL,
        PostTypes.REMOVE_FROM_CHANNEL,
        PostTypes.JOIN_TEAM,
        PostTypes.LEAVE_TEAM,
        PostTypes.ADD_TO_TEAM,
        PostTypes.REMOVE_FROM_TEAM,
    ],
    USER_ACTIVITY_POST_TYPES: [
        PostTypes.ADD_TO_CHANNEL,
        PostTypes.JOIN_CHANNEL,
        PostTypes.LEAVE_CHANNEL,
        PostTypes.REMOVE_FROM_CHANNEL,
        PostTypes.ADD_TO_TEAM,
        PostTypes.JOIN_TEAM,
        PostTypes.LEAVE_TEAM,
        PostTypes.REMOVE_FROM_TEAM,
    ],
};
