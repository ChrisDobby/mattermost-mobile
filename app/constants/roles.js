// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    SYSTEM_GUEST_ROLE: 'system_guest',
    SYSTEM_USER_ROLE: 'system_user',
    SYSTEM_ADMIN_ROLE: 'system_admin',

    TEAM_GUEST_ROLE: 'team_guest',
    TEAM_USER_ROLE: 'team_user',
    TEAM_ADMIN_ROLE: 'team_admin',

    CHANNEL_GUEST_ROLE: 'channel_guest',
    CHANNEL_USER_ROLE: 'channel_user',
    CHANNEL_ADMIN_ROLE: 'channel_admin',
};
