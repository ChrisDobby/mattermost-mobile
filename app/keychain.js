// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {NativeModules} from 'react-native';
import {getGenericPassword, resetGenericPassword, setGenericPassword} from 'react-native-keychain';

import avoidNativeBridge from 'app/utils/avoid_native_bridge';

const {Initialization} = NativeModules;

export async function loadCredentials() {
    const credentials = await avoidNativeBridge(
        () => {
            return Initialization.credentialsExist;
        },
        () => {
            return Initialization.credentials;
        },
        () => {
            return getGenericPassword();
        }
    );

    if (!credentials) {
        return null;
    }

    // The username is made up of the device token and user ID separated by a comma
    const usernameParsed = credentials.username.split(',');

    // The password is made up of the session token and server URL separated by a comma
    const passwordParsed = credentials.password.split(',');

    if (usernameParsed.length !== 2 || passwordParsed.length !== 2) {
        // These credentials are invalid
        return null;
    }

    const [deviceToken, currentUserId] = usernameParsed;
    const [token, url] = passwordParsed;

    return {
        currentUserId,
        deviceToken,
        token,
        url,
    };
}

export async function storeCredentials(deviceToken, currentUserId, token, url) {
    // Only save to keychain if the url and token are set
    if (!url || !token) {
        return;
    }

    const username = `${deviceToken},${currentUserId}`;
    const password = `${token},${url}`;

    await setGenericPassword(username, password);
}

export async function clear() {
    await resetGenericPassword();
}
