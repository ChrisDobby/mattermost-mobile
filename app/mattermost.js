// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Linking, NativeModules, Platform} from 'react-native';
import {Navigation} from 'react-native-navigation';

import {loadMe as loadMeRedux} from 'mattermost-redux/actions/users';

import {resetToChannel, resetToSelectServer} from 'app/actions/navigation';
import {setDeepLinkURL} from 'app/actions/views/root';
import {loadMe} from 'app/realm/actions/user';
import {getAppCredentials} from 'app/init/credentials';
import emmProvider from 'app/init/emm_provider';
import 'app/init/device';
import 'app/init/fetch';
import globalEventHandler from 'app/init/global_event_handler';
import pushNotificationsHandler from 'app/init/push_notifications_handler';
import {registerScreens} from 'app/screens';
import {configureRealmStore, configureAppStore} from 'app/store';
import EphemeralStore from 'app/store/ephemeral_store';
import telemetry from 'app/telemetry';

const {MattermostShare} = NativeModules;
const sharedExtensionStarted = Platform.OS === 'android' && MattermostShare.isOpened;
const reduxStore = configureAppStore();
let realmStore;

const init = async () => {
    const credentials = await getAppCredentials();
    if (EphemeralStore.currentServerUrl) {
        realmStore = configureRealmStore(EphemeralStore.currentServerUrl);
        EphemeralStore.setRealmStoreByServer(EphemeralStore.currentServerUrl, realmStore);
    } else {
        realmStore = configureRealmStore();
    }

    if (EphemeralStore.appStarted) {
        launchApp(credentials);
        return;
    }

    pushNotificationsHandler.configure(reduxStore);
    globalEventHandler.configure({
        reduxStore,
        launchApp,
    });

    registerScreens(reduxStore);

    if (sharedExtensionStarted) {
        EphemeralStore.appStarted = true;
    }

    if (!EphemeralStore.appStarted) {
        launchAppAndAuthenticateIfNeeded(credentials);
    }
};

const launchApp = async (credentials) => {
    telemetry.start([
        'start:select_server_screen',
        'start:channel_screen',
    ]);

    if (credentials) {
        reduxStore.dispatch(loadMeRedux());

        if (realmStore) {
            realmStore.dispatch(loadMe());
        }

        resetToChannel({skipMetrics: true});
    } else {
        resetToSelectServer(emmProvider.allowOtherServers);
    }

    telemetry.startSinceLaunch(['start:splash_screen']);
    EphemeralStore.appStarted = true;
};

const launchAppAndAuthenticateIfNeeded = async (credentials) => {
    await emmProvider.handleManagedConfig(reduxStore);
    await launchApp(credentials);

    if (emmProvider.enabled) {
        if (emmProvider.jailbreakProtection) {
            emmProvider.checkIfDeviceIsTrusted();
        }

        if (emmProvider.inAppPinCode) {
            await emmProvider.handleAuthentication(reduxStore);
        }
    }

    Linking.getInitialURL().then((url) => {
        reduxStore.dispatch(setDeepLinkURL(url));
    });
};

Navigation.events().registerAppLaunchedListener(() => {
    init();

    // Keep track of the latest componentId to appear
    Navigation.events().registerComponentDidAppearListener(({componentId}) => {
        EphemeralStore.addNavigationComponentId(componentId);
    });

    Navigation.events().registerComponentDidDisappearListener(({componentId}) => {
        EphemeralStore.removeNavigationComponentId(componentId);
    });
});
