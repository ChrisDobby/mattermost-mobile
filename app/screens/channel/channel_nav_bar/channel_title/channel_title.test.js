// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {General, Preferences} from 'app/constants';

import ChannelTitle from './channel_title';

jest.mock('react-intl');

describe('ChannelTitle', () => {
    const baseProps = {
        canHaveSubtitle: false,
        channelType: General.OPEN_CHANNEL,
        displayName: 'My Channel',
        hasGuests: false,
        isArchived: false,
        isChannelMuted: false,
        isGuest: false,
        isOwnDM: false,
        onPress: jest.fn(),
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelTitle {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when isOwnDM is true', () => {
        const props = {
            ...baseProps,
            channelType: General.DM_CHANNEL,
            displayName: 'My User',
            isOwnDM: true,
        };
        const wrapper = shallow(
            <ChannelTitle {...props}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
