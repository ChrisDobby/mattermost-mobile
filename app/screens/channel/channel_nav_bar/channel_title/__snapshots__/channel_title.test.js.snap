// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelTitle should match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      My Channel
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": "#ffffff",
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`ChannelTitle should match snapshot when isOwnDM is true 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      <FormattedText
        defaultMessage="{displayName} (you)"
        id="channel_header.directchannel.you"
        values={
          Object {
            "displayName": "My User",
          }
        }
      />
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": "#ffffff",
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
